### DISCUSSION TITLE
incremental function does not work

### DISCUSSION THREAD

**Discussion by isbincApps** [supabase/supabase#8199] [Questions]
Hi, I wrote a function to increment, but sometimes it works and other times it tells me that it doesn't exist.

```
CREATE OR REPLACE FUNCTION incrementbill(mempresa text, mcompany text, mcodigo text)
returns int as
$$
  update configuraciontransacciones
  set ultimonumerodigitado = ultimonumerodigitado + 1
  where empresa = mempresa and company = mcompany and codigo = mcodigo;

  select ultimonumerodigitado from configuraciontransacciones where empresa = mempresa and company = mcompany and 
  codigo = mcodigo;
$$
language sql volatile;
```
this function works, but after some time it throws me this error

```
{message: 'Could not find the public.incrementbill(mcompany, …named json or jsonb parameter in the schema cache', hint: 'If a new function was created in the database with…e and parameters, try reloading the schema cache.'}
```

**Comment by steve-chavez** [comment]
Hello,

Can you share the JS snippet you use to call the function?

**Reply by isbincApps** [reply]
```
          const { data: ultimodigitado, error: errorDoc } = await supabase
                 .rpc('incrementbill', {
                    'mcodigo': headVals.buscar_codigotransaccion.codigo, 
                    'mcompany': '1', 
                    'mempresa': '1'
                  })

```

**Reply by steve-chavez** [reply]
It seems that `headVals.buscar_codigotransaccion.codigo` might be `undefined` sometimes, perhaps you can add a check there?

> {message: 'Could not find the public.incrementbill(mcompany, …named json or jsonb parameter in the schema cache', hint: 'If a new function was created in the database with…e and parameters, try reloading the schema cache.'}

Also, just to double check, could you print the full error message above(there is a `"..."`).
